### ISSUE TITLE
feat: Better animation when scrolling `IonInput` into view (iOS)

### ISSUE THREAD

**Issue by vojto** [ionic-team/ionic-framework#26801]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already include this feature request, without success.

### Describe the Feature Request

When `IonInput` gets focused, it gets scrolled into view, however the animation looks janky compared to native WebView behavior:

**Current `IonInput` animation:**

https://user-images.githubusercontent.com/50943/219069295-b3404b0c-c6d6-4b0d-9003-264e86f42c44.mp4

---

**Native animation:**

(if you don't add `@capacitor/keyboard` and use vanilla `<input />`, this animation will be used)

https://user-images.githubusercontent.com/50943/219069658-c07f6330-1fc2-499f-8753-f4c36ea50887.mp4

---

The `IonInput` animation should look just like the native animation, to provide the native feel of the app.

### Describe the Use Case

Any time input is used, such as login screens, settings, etc.

### Describe Preferred Solution

`IonInput` should animate the same way as native WebView does

### Describe Alternatives

Update `@capacitor/keyboard` to allow programmatic disabling of this code:

```
[nc removeObserver:self.webView name:UIKeyboardWillHideNotification object:nil];
[nc removeObserver:self.webView name:UIKeyboardWillShowNotification object:nil];
[nc removeObserver:self.webView name:UIKeyboardWillChangeFrameNotification object:nil];
[nc removeObserver:self.webView name:UIKeyboardDidChangeFrameNotification object:nil];
```

Code above disables auto-scroll behavior of WebView. As soon as `@capacitor/keyboard` is run, it is added.

One solution would be to allow enabling/disabling of this programmatically - then we could opt into native scrolling on a per-screen basis.

### Related Code

```typescript
export const App: React.FC<AppProps> = () => {
  return (
    <IonApp>
      <IonPage>
        <IonHeader>
          <IonToolbar>
            <IonTitle>Reflect Mobile</IonTitle>
          </IonToolbar>
        </IonHeader>

        <IonContent forceOverscroll={false}>
          <div className="space-y-8 p-8">
            <div className="bg-gray-200 p-8">spacer</div>
            <div className="bg-gray-200 p-8">spacer</div>
            <div className="bg-gray-100 p-8">hello world!</div>
            <div className="bg-gray-100 p-8">
              <IonInput className="w-full text-red-500" value="Hello world!" />
            </div>
          </div>
        </IonContent>
      </IonPage>
    </IonApp>
  )
```

**Comment by liamdebeasi** [comment]
Thanks for the report. Can you help me better understand the request? Are you referring to the smoothness of the animation? Alternatively, are you referring to the time between when the input is tapped and when the "adjustment" animation starts?

**Comment by vojto** [comment]
Sure, the "show" animation in `IonInput` starts too late, after the keyboard is fully visible. If you compare it to the native video, you'll notice the scrolling starts as soon as user taps on the input.

The "hide" animation is not even an animation, it just jumps into the new scroll position.

**Comment by liamdebeasi** [comment]
Thanks for the additional information. The delay for the show animation is going to depend on your Capacitor webview resize configuration. The "native" behavior you noted with an `<input />` is not correct for mobile apps because it causes the header to get pushed offscreen. You can see this in your sample video.

Instead, the content should be scrolled such that the input is not hidden by the keyboard. In order to do this, we need to ensure there is enough scroll space below the input. For example, if the input is at the bottom of the scroll view and the keyboard is 200px tall, we need to scroll by ~200px so the input is not covered by the keyboard. But since the input is at the bottom of the scroll view, we do not have any more room to scroll.

----

The default behavior of your Capacitor app is to resize the entire webview when the keyboard opens. The benefit of this is it ensures we have enough scroll space to scroll the input. The tradeoff here is we need to wait for the webview to finish resizing before we can scroll which does not happen until the keyboard is fully visible. This results in the delay you noted.

The other option is to disable webview resizing (see the "None" option on https://capacitorjs.com/docs/apis/keyboard#keyboardresize). When doing this Ionic adds CSS padding to the bottom of the scroll container to artificially add the scroll space. This allows us to scroll the input as soon as the keyboard begins to open.

There was a bug that caused this behavior to sometimes not work (https://github.com/ionic-team/ionic-framework/issues/18532). However, this is resolved in the Ionic 7 beta.

----

In terms of a path forward, I recommend disabling webview resizing in your Capacitor app and trying the Ionic 7 beta. 

You can disable webview resizing using the Capacitor keyboard plugin: https://capacitorjs.com/docs/apis/keyboard#examples
You can find the Ionic 7 migration guide here: https://ionicframework.com/docs/v7/updating/7-0

**Comment by vojto** [comment]
Thanks for the reply.

I tried Ionic 7 beta + keyboard resizing set to none. Found some issues still, filed as separate issue here: https://github.com/ionic-team/ionic-framework/issues/26803

I think in our use case, it would be much simpler to just use the system behavior. I created a separate issue for capacitor/keyboard: https://github.com/ionic-team/capacitor-plugins/issues/1421

Closing this one.
